### PR TITLE
🧪 Add tests for ParseInfoPkgsFS and ParseInfoPkgs

### PR DESCRIPTION
**Pull request by arran4**
🎯 What: Added tests for the previously untested functions ParseInfoPkgsFS and ParseInfoPkgs in info_pkgs.go.
📊 Coverage: Added tests covering successful parsing, file not found edge cases, and file I/O errors (e.g., trying to read a directory as a file) using both fstest.MapFS and physical t.TempDir().
✨ Result: Test coverage has been increased, ensuring the safety of future refactors to this file-parsing functionality.

---
*PR created automatically by Jules for task [14327862192856585082](https://jules.google.com/task/14327862192856585082) started by @arran4*

### CODE REVIEW

File: info_pkgs_test.go [hunk 1]
```diff
@@ -59,3 +62,93 @@ func TestSerializeInfoPkgs(t *testing.T) {
...
+				t.Errorf("ParseInfoPkgsFS() = %v, want %v", results, tt.expected)
+			}
+		})
+	}
+}
+
+func TestParseInfoPkgs(t *testing.T) {
+	t.Run("existing file", func(t *testing.T) {
+		tempDir := t.TempDir()
+		path := tempDir + "/info_pkgs"
```
Comment (arran4) [inline info_pkgs_test.go]: @jules please adhere to not writing to disks or using disks during tests use refactor for dependency injection
